pkg/rekor/identity.go

```diff
@@ -313,3 +354,23 @@ func certMatchesPolicy(cert *x509.Certificate, expectedSub string, expectedIssue
 	}
 	return subjectMatches && issuerMatches, matchedSub, issuer, nil
 }
+
+// oidMatchesPolicy returns if a certificate contains both a given OID field and a matching value associated with that field
+// if true, it returns the OID extension and extension value that were matched on
+func oidMatchesPolicy(cert *x509.Certificate, oid asn1.ObjectIdentifier, extensionValues []string) (bool, asn1.ObjectIdentifier, string, error) {
+	extValue, err := getExtension(cert, oid)
+	if err != nil {
+		return false, nil, "", fmt.Errorf("error getting extension value: %w", err)
+	}
+	if extValue == "" {
+		return false, nil, "", nil
+	}
```

Comment:
Should we move this check first in the function? We don't need to `getExtension` if we can return an empty/default tuple

Reply:
i think we're checking if the extension field returned by getExtension is empty and returning on the empty case here- i can add checks for if the input args of oid/extensionValues are empty if that would be preferable?

Reply:
Sorry, I misread, somehow I confused `extValue` with `extensionValues`